packages/chain-bridge/src/router/base/BridgeRouterValidationBase.sol

```diff
@@ -0,0 +1,97 @@
+import {ICrossChainReceiver} from "../../interfaces/ICrossChainReceiver.sol";
+import {IERC165} from "@openzeppelin/contracts/interfaces/IERC165.sol";
+import {IBridgeRouter} from "../../interfaces/IBridgeRouter.sol";
+
+/**
+ * @title BridgeRouterValidationBase
+ * @notice Abstract base contract providing validation utilities for BridgeRouter operations
+ * @dev Contains all validation logic extracted from BridgeRouter
+ */
+abstract contract BridgeRouterValidationBase {
+    /*//////////////////////////////////////////////////////////////
+                        VALIDATION FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Internal function to validate transfer parameters
+     * @param params Parameters to validate
+     */
+    function _validateTransferParams(
+        BridgeTypes.ExecuteTransferParams calldata params
+    ) internal pure {
+        if (
+            params.amount == 0 ||
+            params.target == address(0) ||
+            params.originator == address(0) ||
+            params.asset == address(0) ||
+            params.refundAddress == address(0) ||
+            params.destinationChainId == 0
+        ) revert IBridgeRouter.InvalidParams();
+    }
+
+    /**
+     * @dev Internal function to validate send message parameters
+     * @param params Parameters to validate
+     */
+    function _validateSendMessageParams(
+        BridgeTypes.ExecuteSendMessageParams calldata params
+    ) internal pure {
+        if (
+            params.target == address(0) ||
+            params.originator == address(0) ||
+            params.destinationChainId == 0 ||
+            params.refundAddress == address(0) ||
+            params.message.length == 0
+        ) {
+            revert IBridgeRouter.InvalidParams();
+        }
+    }
+
+    /**
+     * @dev Internal function to validate originator
+     * @param originator The originator address
+     */
+    function _validateOriginator(address originator) internal view {
+        if (originator != msg.sender) revert IBridgeRouter.InvalidOriginator();
+    }
+
+    /**
+     * @dev Internal function to validate if an adapter supports a specific operation type
+     * @param adapter The adapter address to validate
+     * @param operationType The type of operation to check support for
+     */
+    function _validateAdapterSupportsOperation(
+        address adapter,
+        BridgeTypes.OperationType operationType
+    ) internal view {
+        if (adapter == address(0)) revert IBridgeRouter.NoSuitableAdapter();
+        if (!IBridgeAdapter(adapter).supportsOperation(operationType)) {
+            revert IBridgeRouter.UnsupportedAdapterOperation();
+        }
+    }
+
+    /**
+     * @dev Ensures `receiver` is a contract that supports `ICrossChainReceiver` via ERC165
+     *      Reverts with `InvalidParams` otherwise.
+     * @param receiver The address to validate
+     */
+    function _requireReceiverIsCrossChainReceiver(
```

Comment:
Why the change of nomenclature here? Shouldn't it be `_validateReceiver` and just let the logic say what it means to validate it? Or call it `_validateCrossChainReceiver`?